qcodes_contrib_drivers/drivers/Bilt/ITest.py

```diff
@@ -0,0 +1,475 @@
+            v_set: The target voltage
+        """
+        chan_id = self.chan_to_id(chan)
+        self.write('{}VOLT {:.8f}'.format(chan_id, v_set))
+        self.write(chan_id + 'TRIG:INPUT:INIT')
+        if self._get_chan_synchronous(chan):
+            v = self._get_voltage(chan)
+            while abs(v_set - v)>=self._synchronous_threshold:
+                sleep(self._synchronous_delay)
+                v = self._get_voltage(chan)
+
+
+    def _get_voltage(self, chan:int) -> float:
+        """
+        Get cmd for the chXX_v parameter
+        
+        Args:
+            chan: The 1-indexed channel number
+
+        Returns:
+            Voltage
+        """
+        chan_id = self.chan_to_id(chan)
+        
+        return float(self.ask('{}MEAS:VOLT?'.format(chan_id)))
+
+
+    def _get_current(self, chan:int) -> float:
+        """
+        Get cmd for the chXX_i parameter
+        
+        Args:
+            chan: The 1-indexed channel number
+
+        Returns:
+            Current
+        """
+        chan_id = self.chan_to_id(chan)
+        
+        return float(self.ask('{}MEAS:CURR?'.format(chan_id)))
+
+
+    def _set_ramp_slope(self, chan:int,
+                              slope:float) -> None:
+        """
+        Set slope of chXX for ramp mode
+        
+        Args:
+            chan The 1-indexed channel number
+            slope Slope of chXX in V/ms
+        """
+        chan_id = self.chan_to_id(chan)
+        self.write('{}VOLT:SLOP {:.8f}'.format(chan_id, slope))
+
+
+    def _get_ramp_slope(self, chan:int) -> str:
+        """
+        Get slope of chXX
+        
+        Args:
+            chan: The 1-indexed channel number
+            
+        Returns:
+            chXX_slope parameter
+        """
+        chan_id = self.chan_to_id(chan)
+        return self.ask('{}VOLT:SLOP?'.format(chan_id))
+
+
+    def _set_output_function(self, chan:int,
+                                   outf: Union[int, str]) -> None:
+        """
+        Set how to perform output voltage update
+        
+        Args:
+            chan: The 1-indexed channel number
+            ouf: Mode
+        """
+        chan_id = self.chan_to_id(chan)
+        
+        if isinstance(outf,(int,float)):
+            mode = str(outf)
+        else:
+            if outf=='exp':
+                mode = '0'
+            elif outf=='ramp':
+                mode = '1'
+        
+        self.write(chan_id + 'trig:input ' + mode)
+
+
+    def _get_output_function(self, chan:int) -> str:
+        """
+        Get output volage update  function
+        
+        Args:
+            chan: The 1-indexed channel number
+        
+        Returns:
+            mode
+        """
+        chan_id = self.chan_to_id(chan)
+        mode = int(self.ask(chan_id + 'TRIG:INPUT?'))
+        if mode == 0:
+            return 'exp'
+        elif mode == 1:
+            return 'ramp'
+        else:
+            return 'ERROR'
```

Comment:
shall this raise a proper exception with useful error message? :) otherwise it's less likely that users will understand what is wrong :) something like, `raise ValueError(f"Got unexpected output function mode {mode}")`